Source/WebCore/platform/graphics/ImageBuffer.cpp

```diff
@@ -32,25 +32,64 @@
 #include "GraphicsContext.h"
 #include "HostWindow.h"
 #include "PlatformImageBuffer.h"
+#include <wtf/text/Base64.h>
+
+#if USE(CG)
+#include "ImageBufferUtilitiesCG.h"
+#endif
+#if USE(CAIRO)
+#include "ImageBufferUtilitiesCairo.h"
+#endif
 
 namespace WebCore {
 
 static const float MaxClampedLength = 4096;
 static const float MaxClampedArea = MaxClampedLength * MaxClampedLength;
 
+static RefPtr<ImageBuffer> copyImageBuffer(Ref<ImageBuffer> source, PreserveResolution preserveResolution)
```

Comment:
Can we implement ImageBuffer::clone() by using this function? Also can we use the same naming: ImageBuffer::clone() and cloneImageBuffer() or ImageBuffer::copy() and copyImageBuffer()?

Reply:
Use copyImageBuffer() in clone() now.